include/vcpkg/base/files.h

```diff
@@ -231,6 +208,41 @@ namespace vcpkg
         int put(int c) const noexcept { return ::fputc(c, m_fs); }
     };
 
+    struct ExclusiveFileLock
```

Comment:
At a glance, this seems like a lot more code than

```c++
std::unique_ptr<IExclusiveFileLock>
```

Reply:
I could do that too. This version doesn't allocate, and that one would, but I doubt it matters.

Reply:
Thinking again I think this would be a good simplification since it would let me merge Xxx_helper with Xxx_unlocker. I already have 2 approvals for this one and I still think it's an improvement so I'm going to merge it, but preparing a change to apply this now.